.github/workflows/scripts/scanner-update-nvd.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+
+import os
+import json
+from datetime import datetime, timedelta
+import logging
+from urllib import request, parse
+import time
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s: %(levelname)s: %(message)s")
+
+BASE_URL = os.getenv('SCANNER_NVD_URL', "https://services.nvd.nist.gov/rest/json/cves/2.0")
+API_KEY = os.getenv('SCANNER_NVD_API_KEY', None)
+
+
+def get_dates(year, interval):
+    if interval > 12:
+        interval = 12
+    for month in range(1, 13, interval):
+        start = datetime(year, month, 1)
+
+        # Handling for months greater than 12 (for intervals like 4, 6 months, etc.)
+        end_month = month + interval
+        end_year = year
+        if end_month > 12:
+            end_month -= 12
+            end_year += 1
+
+        end = datetime(end_year, end_month, 1) - timedelta(days=1)
+        yield start.strftime("%Y-%m-%dT00:00:00.000"), end.strftime("%Y-%m-%dT23:59:59.999")
+
+def fetch_data(start, end):
+    if API_KEY is None:
```

Comment:
surprised this is not allowed. We allow the API_KEY to be `None` until we get here. I think it's ok if we don't use the API_KEY, though, since we know we will use it, we should enforce it at the beginning of `main` so we don't waste CPU cycles just to fail here